src/sky/pro/StringListImpl.java

```diff
@@ -0,0 +1,224 @@
+            array[--size] = null;
+        }
+    }
+
+    // Удаление элемента по индексу.
+    // Вернуть удаленный элемент
+    // или исключение, если подобный
+    // элемент отсутствует в списке.
+    @Override
+    public String remove(int index) {
+        checkIndex(index);
+        String element = get(index);
+        removeElement(index);
+        return element;
+    }
+
+    // Проверка на существование элемента.
+    // Вернуть true/false;
+    @Override
+    public boolean contains(String element) {
+        checkNotNull(element);
+        boolean result = false;
+        for (int i = 0; i < index; i++) {
+            if (array[i].equals(element)) {
+                result = true;
+                break;
+            }
+        }
+        return result;
+    }
+
+    // Поиск элемента.
+    // Вернуть индекс элемента
+    // или -1 в случае отсутствия.
+    @Override
+    public int indexOf(String element) {
+        int index = -1;
+        checkNotNull(element);
+        for (int i = 0; i < size; i++) {
+            if (array[i].equals(element)) {
+                index = i;
+                break;
+            }
+        }
+        return index;
+    }
+
+    // Поиск элемента с конца.
+    // Вернуть индекс элемента
+    // или -1 в случае отсутствия.
+    @Override
+    public int lastIndexOf(String element) {
+        int index = -1;
+        checkNotNull(element);
+        for (int i = size - 1; i >= 0; i--) {
+            if (array[i].equals(element)) {
+                index = i;
+                break;
+            }
+        }
+        return index;
+    }
+
+    // Получить элемент по индексу.
+    // Вернуть элемент или исключение,
+    // если выходит за рамки фактического
+    // количества элементов.
+    @Override
+    public String get(int index) {
+        checkIndex(index);
+        return array[index];
+    }
+
+    // Сравнить текущий список с другим.
+    // Вернуть true/false или исключение,
+    // если передан null.
+
+    @Override
+    public boolean equals(StringList otherList) {
+        if (otherList == null) {
+            return false;
+        }
+        if (size != otherList.size()) {
+            return false;
+        }
+        for (int i = 0; i < size; i++) {
+            if (!get(i).equals(otherList.get(i))) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    // Вернуть фактическое количество элементов.
+    public int size() {
+        return size;
+    }
+
+    // Вернуть true,
+    // если элементов в списке нет,
+    // иначе false.
+    public boolean isEmpty() {
+        return size == 0;
+    }
+
+    // Удалить все элементы из списка.
+    @Override
+    public void clear() {
+        for (int i = 0; i < size; i++) {
+            array[i] = null;
+        }
+    }
+
+    // Создать новый массив
+    // из строк в списке
+    // и вернуть его.
+    @Override
+    public String[] toArray() {
+        return Arrays.copyOf(array, array.length);
```

Comment:
```suggestion
        return Arrays.copyOf(array, array.index);
```
пустые элементы неинтересны снаружи